starlite/testing/async_test_client/transport.py

```diff
@@ -0,0 +1,187 @@
+    from starlite.types import (
+        HTTPDisconnectEvent,
+        HTTPRequestEvent,
+        Message,
+        Receive,
+        ReceiveMessage,
+        Send,
+        WebSocketScope,
+    )
+
+
+class ConnectionUpgradeException(Exception):
+    def __init__(self, session: "WebSocketTestSession") -> None:
+        self.session = session
+
+
+class SendReceiveContext(TypedDict):
+    request_complete: bool
+    response_complete: Event
+    raw_kwargs: Dict[str, Any]
+    response_started: bool
+    template: Optional[str]
+    context: Optional[Any]
+
+
+class AsyncTestClientTransport(AsyncBaseTransport):
+    def __init__(
+        self,
+        client: "AsyncTestClient",
+        raise_server_exceptions: bool = True,
+        root_path: str = "",
+    ) -> None:
+        self.client = client
+        self.raise_server_exceptions = raise_server_exceptions
+        self.root_path = root_path
+
+    @staticmethod
+    def create_receive(request: "Request", context: SendReceiveContext) -> "Receive":
+        async def receive() -> "ReceiveMessage":
+            if context["request_complete"]:
+                if not context["response_complete"].is_set():
+                    await context["response_complete"].wait()
+                disconnect_event: "HTTPDisconnectEvent" = {"type": "http.disconnect"}
+                return disconnect_event
+
+            body = cast("Union[bytes, str, GeneratorType]", (request.read() or b""))
+            request_event: "HTTPRequestEvent" = {"type": "http.request", "body": b"", "more_body": False}
+            if isinstance(body, GeneratorType):  # pragma: no cover
+                try:
+                    chunk = body.send(None)
+                    request_event["body"] = chunk if isinstance(chunk, bytes) else chunk.encode("utf-8")
+                    request_event["more_body"] = True
+                except StopIteration:
+                    context["request_complete"] = True
+            else:
+                context["request_complete"] = True
+                request_event["body"] = body if isinstance(body, bytes) else body.encode("utf-8")
+            return request_event
+
+        return receive
+
+    @staticmethod
+    def create_send(request: "Request", context: SendReceiveContext) -> "Send":
+        async def send(message: "Message") -> None:
+            if message["type"] == "http.response.start":
+                if context["response_started"]:
+                    raise AssertionError('Received multiple "http.response.start" messages.')
+                context["raw_kwargs"]["status_code"] = message["status"]
+                context["raw_kwargs"]["headers"] = [
+                    (k.decode("utf-8"), v.decode("utf-8")) for k, v in message.get("headers", [])
+                ]
+                context["response_started"] = True
+            elif message["type"] == "http.response.body":
+                if not context["response_started"]:
```

Comment:
this is the statement where flake8 has problems, even after changing the `assert statement` to `if statement` and running the flake8 pre-commit hook locally ( which passes ) it still has some issues here in the ci
````diff
- assert context["response_started"] , 'Received "http.response.body" without "http.response.start".'
+ if not context["response_started"]:
+ raise AssertionError('Received "http.response.body" without "http.response.start".')
````